src/pages/[platform]/build-a-backend/data/query-data/index.mdx

```diff
@@ -245,6 +266,86 @@ const fetchPosts = async () => {
 }
 ```
 
+</InlineFilter>
+
+<InlineFilter filters={['vue']}>
+```ts
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { generateClient, type SelectionSet } from 'aws-amplify/data';
+import type { Schema } from '../../../amplify/data/resource';
+
+const client = generateClient<Schema>();
+
+const selectionSet = ['content', 'blog.author.*', 'comments.*'] as const;
```

Comment:
```suggestion
const selectionSet = ['content', 'blog.author.*', 'comments.*'] as const satisfies readonly (<some-selection-set-type??>)[];
```

this isn't a huge deal but it'd be nice to narrow this type if possible

Reply:
Can't figure out a way to narrow this. We don't expose a selection set type from `Schema['model']['...']`.